src/D2L.CodeStyle.Analyzers/Language/AttributeAliasesAnalyzer.cs

```diff
@@ -0,0 +1,115 @@
+﻿using System.Collections.Immutable;
+using D2L.CodeStyle.Analyzers.Extensions;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace D2L.CodeStyle.Analyzers.Language {
+
+	[DiagnosticAnalyzer( LanguageNames.CSharp )]
+	internal sealed class AttributeAliasesAnalyzer : DiagnosticAnalyzer {
+
+		public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(
+			Diagnostics.AliasingAttributeNamesNotSupported
+		);
+
+		public override void Initialize( AnalysisContext context ) {
+			context.EnableConcurrentExecution();
+			context.ConfigureGeneratedCodeAnalysis( GeneratedCodeAnalysisFlags.Analyze | GeneratedCodeAnalysisFlags.ReportDiagnostics );
+			context.RegisterCompilationStartAction( CompilationStart );
+		}
+
+		private void CompilationStart( CompilationStartAnalysisContext context ) {
+
+			context.RegisterSyntaxNodeAction(
+					c => AnalyzeAttribute( c, (AttributeSyntax)c.Node ),
+					SyntaxKind.Attribute
+				);
+		}
+
+		private void AnalyzeAttribute(
+				SyntaxNodeAnalysisContext context,
+				AttributeSyntax attribute
+			) {
+
+			// if it's not an identifier, then it's qualified and not aliasing the attribute name
+			if( attribute.Name is not IdentifierNameSyntax attributeName  ) {
+				return;
+			}
+
+			IEnumerable<IdentifierNameSyntax> usingAliases = GetUsingAliases( attribute );
+			foreach( IdentifierNameSyntax usingAlias in usingAliases ) {
+
+				if( !IsEquivalentToUsingAlias( attributeName, usingAlias ) ) {
+					continue;
+				}
+
+				Diagnostic d = Diagnostic.Create(
+						Diagnostics.AliasingAttributeNamesNotSupported,
+						attribute.GetLocation()
+					);
+
+				context.ReportDiagnostic( d );
+				return;
+			}
+		}
+
+		private static bool IsEquivalentToUsingAlias(
+				IdentifierNameSyntax attributeIdentifier,
+				IdentifierNameSyntax usingAliasIdentifier
+			) {
+
+			string attributeName = attributeIdentifier.ToString();

```

Comment:
nit: `ToString` in every iteration

Reply:
I think the following is substantially true,

    `number of files with attributes` > `number of files with attributes & more than 1 alias`

So much so that formatting the string ahead of time is probably extra work in the grand scheme of things.